qiskit_experiments/warnings.py

```diff
@@ -0,0 +1,245 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""A collection of typical warnings."""
+
+import functools
+import warnings
+
+from typing import Callable, Optional, Type, Dict
+
+
+def deprecated_logic(
+    version_removed: Optional[str] = None,
```

Comment:
nitpick but "removed" is past-tense, and this is talking about a future version. Should be `last_version` or something like that and say `will be removed after {last_version}`